ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -372,12 +371,17 @@ public void onCompletion(String result, Exception exception) {
               router.deleteBlob(result, deleteCallback);
               break;
             default:
-              throw new IllegalStateException("Unrecognized RestMethod: " + restMethod);
+              exception = new IllegalStateException("Unrecognized RestMethod: " + restMethod);
           }
+        } catch (Exception e) {
+          exception = e;
         }
-      } finally {
-        callbackTracker.markCallbackProcessingEnd();
       }
+
+      if (exception != null) {
+        submitResponse(restRequest, restResponseChannel, null, exception);
+      }
+      callbackTracker.markCallbackProcessingEnd();
```

Comment:
Not for this part but for Line 475 and Line 487. It seems a variable is assigned to itself.